plugin/core/sessions.py

```diff
@@ -428,7 +428,7 @@ def get_plugin(name: str) -> Optional[Type[AbstractPlugin]]:
 
 class Session(Client):
 
-    def __init__(self, manager: Manager, logger: Logger, workspace_folders: List[WorkspaceFolder],
+    def __init__(self, manager: Manager, loggers: List[Logger], workspace_folders: List[WorkspaceFolder],
```

Comment:
I'm not a fan of this list-of-loggers approach :/ We could just as well have a ListLogger class that delegates all of the methods to all of its subloggers.

(untested)
```python
class ListLogger(Logger):
    def __init__(self) -> None:
        self._loggers = []  # type: List[Logger]

    def append(self, logger: Logger) -> None:
        self._loggers.append(logger)

    def _foreach(self, method: str, *args: Any) -> None:
        for logger in self._loggers:
            getattr(logger, method)(*args)

    def outgoing_response(self, *args: Any) -> None:
        self._foreach("outgoing_response", *args)

    def outgoing_error_response(self, *args: Any) -> None:
        self._foreach("outgoing_error_response", *args)

    def outgoing_request(self, *args: Any) -> None:
        self._foreach("outgoing_response", *args)

    def outgoing_notification(self, *args: Any) -> None:
        self._foreach("outgoing_notification", *args)

    def incoming_response(self, *args: Any) -> None:
        self._foreach("incoming_response", *args)

    def incoming_request(self, *args: Any) -> None:
        self._foreach("incoming_request", *args)

    def incoming_notification(self, *args: Any) -> None:
        self._foreach("incoming_notification", *args)
```

Also untested _create_loggers modification:
```python
    def _create_loggers(self, config_name: str) -> Logger:
        logger_map = {
            "panel": PanelLogger,
            "remote": RemoteLogger,
        }
        loggers = []
        for logger_type in settings.log_server:
            if logger_type not in logger_map:
                debug("Invalid logger type ({}) specified for log_server settings".format(logger_type))
                continue
            loggers.append(logger_map[logger_type])
        if len(loggers) == 0:
            return ListLogger()  # logs nothing
        elif len(loggers) == 1:
            return loggers[0](self, config_name)
        else:
            logger = ListLogger()
            for logger_type in loggers:
                logger.append(logger_type(self, config_name))
            return logger
```

Reply:
To be clear, uou are not a fan of this implementation but you are OK with user-facing changes to `log_server` setting?

I'm OK with your suggestion although all I think of when seeing it is "more lines of code".

Reply:
Apologies for the confusion. I'm certainly OK with user-facing changes to log_server. The remote logging is pretty cool. I just don't agree with changing Session and Client for this. My proposed changes would keep Client as-is. It's perhaps more lines of code, but the functionality (doing stuff for a list-of-loggers) is contained in a single object.